src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IUnaryOperatorExpression.cs

```diff
@@ -0,0 +1,2718 @@
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.FloatingMinus) (OperationKind.UnaryOperatorExpression, Type: System.Double)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Double)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Minus_System_Boolean()
+        {
+            string source = @"
+class A
+{
+    System.Boolean Method()
+    {
+        System.Boolean i = default(System.Boolean);
+        return /*<bind>*/-i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Boolean)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Type_Minus_System_Object()
+        {
+            string source = @"
+class A
+{
+    System.Object Method()
+    {
+        System.Object i = default(System.Object);
+        return /*<bind>*/-i/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.Invalid) (OperationKind.UnaryOperatorExpression, Type: System.Object, IsInvalid)
+  ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Object)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Method_Plus_System_SByte()
+        {
+            string source = @"
+class A
+{
+    System.SByte Method()
+    {
+        System.SByte i = default(System.SByte);
+        return /*<bind>*/+Method()/*</bind>*/;
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IUnaryOperatorExpression (UnaryOperationKind.IntegerPlus) (OperationKind.UnaryOperatorExpression, Type: System.Int32)
+  IConversionExpression (ConversionKind.CSharp, Implicit) (OperationKind.ConversionExpression, Type: System.Int32)
+    IInvocationExpression ( System.SByte A.Method()) (OperationKind.InvocationExpression, Type: System.SByte)
+      Instance Receiver: IInstanceReferenceExpression (InstanceReferenceKind.Implicit) (OperationKind.InstanceReferenceExpression, Type: A)
+";
+            VerifyOperationTreeForTest<PrefixUnaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+
+        [Fact]
+        public void Test_UnaryOperatorExpression_Method_Plus_System_Byte()
+        {
+            string source = @"
+class A
+{
+    System.Byte Method()
+    {
+        System.Byte i = default(System.Byte);
+        return /*<bind>*/+Method()/*</bind>*/;
```

Comment:
since Operand can be any expression, I wanted to have at least 2 different kind of expression there. if you have better expression to use there, let me know.